crates/forge_permission_resolver/src/permissions_cache.rs

```diff
@@ -0,0 +1,261 @@
+use regex::Regex;
+use std::collections::HashMap;
+use std::env;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use crate::permissions_resolver::PermissionHashMap;
+use crate::serde::{SerializablePermissionHashMap, ToStringMap};
+use std::fs::{self, File};
+use std::io::{Read, Write};
+use tracing::debug;
+
+const CACHE_EXPIRATION: Duration = Duration::from_secs(60 * 60 * 24 * 7); // 1 week
+
+fn default_cache_path() -> PathBuf {
+    if let Ok(home) = env::var("HOME") {
+        PathBuf::from(home).join(".cache/fsrt")
+    } else {
+        PathBuf::new()
+    }
+}
+
+/// Struct to hold cache-related settings
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct CacheConfig {
+    use_cache: bool,
+    cache_path: Option<PathBuf>,
+    ttl: Duration,
+}
+
+impl CacheConfig {
+    pub fn new(use_cache: bool, cache_path: Option<PathBuf>) -> Self {
+        CacheConfig {
+            use_cache,
+            cache_path: Some(cache_path.unwrap_or_else(default_cache_path)),
+            ttl: CACHE_EXPIRATION,
+        }
+    }
+
+    pub fn use_cache(&self) -> bool {
+        self.use_cache
+    }
+
+    pub fn cache_path(&self) -> &PathBuf {
```

Comment:
```suggestion
    pub fn cache_path(&self) -> Option<&Path> {
```

That way, you can get rid of `use_cache()` and avoid bugs by encoding the absence of the cache in the state of the data itself.